### PR TITLE
Improve many SwitchProducer unit tests

### PR DESCRIPTION
**Pull request by makortel**
#### PR description:

While looking the SwitchProducer unit tests I realized many of them don't actually test the behavior they should test. This PR makes many improvements to the tests.

#### PR validation:

Framework unit tests run.

### CODE REVIEW

**Comment by makortel** [comment]
@cmsbuild, please test

File: FWCore/Integration/test/testSwitchProducerPath_cfg.py [hunk 1]
```diff
@@ -57,6 +57,24 @@ def __init__(self, **kargs):
                                                  cms.PSet(type = cms.string("edmtestIntProduct"), fromProductInstance = cms.string("foo"), toProductInstance = cms.string("other"))))
 )
 
+# SwitchProducer of EDAlias in a Path causes alone should not cause aliased-for EDProducers to run
```
Comment (makortel) [inline FWCore/Integration/test/testSwitchProducerPath_cfg.py:60]:
```suggestion
# SwitchProducer of EDAlias in a Path causes alone should cause aliased-for EDProducers to run
```

**Comment by makortel** [comment]
@cmsbuild, please test

**Comment by makortel** [comment]
+1

**Comment by qliphy** [comment]
+1

**Pull request merged**